keep/api/core/db.py

```diff
@@ -958,6 +962,32 @@ def get_last_alerts(
                 )
                 .subquery()
             )
+            
+        if upper_timestamp is not None and lower_timestamp is not None:
+            subquery = (
+                session.query(subquery)
+                .filter(
+                    subquery.c.max_timestamp < upper_timestamp,
+                    subquery.c.max_timestamp >= lower_timestamp
+                )
+                .subquery()
+            )
+        elif upper_timestamp is not None:
+            subquery = (
+                session.query(subquery)
+                .filter(
+                    subquery.c.max_timestamp < upper_timestamp
+                )
+                .subquery()
+            )
+        elif lower_timestamp is not None:
+            subquery = (
+                session.query(subquery)
+                .filter(
+                    subquery.c.max_timestamp >= lower_timestamp
+                )
+                .subquery()
+            )
```

Comment:
can't timeframe do that?

Reply:
Timeframe doesn't allow us to set both upper and lower bounds